src/Xamarin.Android.Build.Tasks/Tasks/RemoveDirFixed.cs

```diff
@@ -79,6 +79,13 @@ public override bool RunTask ()
 					} else {
 						Log.LogUnhandledException (TaskPrefix, ex);
 					}
+				} catch (IOException ex) {
+					try {
+						DeleteDirectoryManually (fullPath);
+					} catch (Exception inner) {
+						Log.LogUnhandledException (TaskPrefix, ex);
+						Log.LogUnhandledException (TaskPrefix, inner);
+					}
```

Comment:
It looks like the plain `<RemoveDir/>` task wouldn't have this issue:
* https://github.com/dotnet/msbuild/blob/2f39510e2abb49b27dcb11122088a678c51288cc/src/Tasks/RemoveDir.cs#L84

It handles the readonly flag on files.

Should we consider using the built-in one instead? Or maybe extending it? Maybe it works in 2024?

Reply:
Its the `Directory not empty` issue we get not `unauthorised`. I'm not sure `RemoveDir` will help. Since `Directory.Delete` can return a `Directory not empty` exception.

Reply:
I can try and see if it works though.

Reply:
This is why we have this task in the first place

```
// The RemoveDir task that ships with Mono 2.10 doesn't work for
// recursive directories, so we ship this one until Mono 2.12 is
// the norm.
```